### PR TITLE
Update LM sparse example

### PR DESCRIPTION
**Pull request by wang-chen**
This pull request updates the example in the `step` method docstring of `pypose/optim/optimizer.py` to demonstrate sparse Levenberg-Marquardt (LM) optimization for bundle adjustment using reprojection error, instead of the previous pose graph optimization example. The new example also modernizes the code to use current APIs and provides a more realistic scenario relevant to computer vision and robotics.

**Documentation and Example Updates:**

* Replaces the pose graph optimization example with a bundle adjustment example that minimizes reprojection error using sparse LM, reflecting a more common and practical use case.
* Updates the example to use `psjac` instead of `parallel_for_sparse_jacobian` for marking sparse Jacobian computation, and demonstrates the use of current `pypose` optimizer components such as `TrustRegion`, `PCG`, and `StopOnPlateau`.
* Provides a more comprehensive and realistic workflow, including pose and point initialization, optimizer setup, and a scheduler for early stopping, making the documentation more helpful for users.

### CODE REVIEW

**Comment by zitongzhan** [comment]
A drawback of the minimum example is that it doesn't show the usage of non-optimizable parameter, `self.register_buffer('root', root)`. Is that needed?

**Comment by wang-chen** [comment]
> A drawback of the minimum example is that it doesn't show the usage of non-optimizable parameter, `self.register_buffer('root', root)`. Is that needed?

Doc don't provide everything, how about show it in example?

**Review by zitongzhan — APPROVED** [review]
Checked that docs render without issue

File: pypose/optim/optimizer.py [hunk 1]
```diff
@@ -563,63 +563,67 @@ def step(self, input, target=None, weight=None):
...
+            >>>
+            >>> while scheduler.continual():
+            ...     loss = optimizer.step(inputs)
+            ...     scheduler.step(loss)
+            StopOnPlateau on step 1 Loss 4.467577e+01 --> Loss 7.737489e+00.
+            StopOnPlateau on step 2 Loss 7.737489e+00 --> Loss 2.681350e+00.
+            StopOnPlateau on step 3 Loss 2.681350e+00 --> Loss 1.192892e+00.
+            StopOnPlateau on step 4 Loss 1.192892e+00 --> Loss 5.546592e-01.
+            StopOnPlateau on step 5 Loss 5.546592e-01 --> Loss 5.288146e-01.
+            StopOnPlateau: Maximum steps reached, Quitting..
```
[the remark this answers is not in the document]
Comment (zitongzhan) [reply pypose/optim/optimizer.py:626]: No longer needed
